src/auditor/auditor.py

```diff
@@ -41,6 +40,46 @@ def retry(worker, verbose=True, tries=1):
             raise error
 
 
+#: TODO: Modify structure so each check/fix method returns its values directly rather than putting them inside
+#: the dictionary, and then log results at the end of each for item in collection iteration.
+def log_report(report_dict, report_file, separator='|', rotate_count=18):
+    """
+    Logs a nested dictionary to a rotating csv file via the logging module.
+
+    report_dict:    Nested dictionary in the form of {Unique ID: {field1:data, field2:data, field3:data},
+                    Unique ID: {field1:data, field2:data, field3:data}, ... }.
+    report_file:    Base log file name, as a string. Automatically rotated by a logging RotatingFileHandler on each
+                    call.
+    separator:      The character used as a csv delimiter. Default to '|' to avoid common conflicts with text data.
+    rotate_count:   The number of files to save before RotatingFileHandler deletes old reports. Defaults to 2.5 weeks.
+
+    """
+    #: Set up a rotating file handler for the report log
+    report_path = Path(report_file)
+    report_logger = logging.getLogger('audit_report')
+    report_handler = logging.handlers.RotatingFileHandler(report_path, backupCount=rotate_count)
+    report_handler.doRollover()  #: Rotate the log on each run
+    report_handler.setLevel(logging.DEBUG)
+    report_logger.addHandler(report_handler)
+    report_logger.setLevel(logging.DEBUG)
+
```

Comment:
**question**: does all of this need to happen every time log_report is called? It seems to me that the logging should be setup once outside of method. This method might be doing too many things?

Reply:
As it's written, `log_report` is only called at most once (at the end of either `check_items` or `fix_items`). If/when I modify the report logging to write at the end of each item's iteration, rather than after the loop, then yes, I'll need to pull this out.